js/hang-ui/src/Components/publish/FileSourceButton.tsx

```diff
@@ -0,0 +1,40 @@
+import { createSignal, useContext } from 'solid-js';
+import { PublishUIContext } from './PublishUIContextProvider';
+
+export default function FileSourceButton() {
+    const [fileInputRef, setFileInputRef] = createSignal<
+        HTMLInputElement | undefined
+    >();
+    const context = useContext(PublishUIContext);
+    const onClick = () => fileInputRef()?.click();
+    const onChange = (event: Event) => {
+        const castedInputEl = event.target as HTMLInputElement;
+        const file = castedInputEl.files?.[0];
+
+        if (file) {
+            context?.setFile(file);
```

Comment:
alternatively, we could expose a setter on the HangPublish element and set it like that.